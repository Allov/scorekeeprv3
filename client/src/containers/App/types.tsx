export interface IUser {
  id: string,
  name: string,
}

export interface IGame {
  shareId: string,
  name: string,
}

export interface IStore {
    me?: IUser,
    game?: IGame,
}
