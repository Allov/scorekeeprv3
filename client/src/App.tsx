import * as React from 'react';

import logo from './logo.svg';

interface IAppProps {
  className?: string,
}

class App extends React.Component<IAppProps, {}> {
  public render() {
    return (
      <div className={this.props.className}>
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">Welcome to React</h1>
          </header>
          <p className="App-intro">
            To get started, edit <code>src/App.tsx</code> and save to reload.
        </p>
        </div>
      </div>
    );
  }
}

export default App;
