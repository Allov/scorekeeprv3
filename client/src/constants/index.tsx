export const CREATE_GAME = 'CREATE_GAME';
export type CREATE_GAME = typeof CREATE_GAME;

export const FETCH_HELLOWORLD = 'FETCH_HELLOWORLD';
export type FETCH_HELLOWORLD = typeof FETCH_HELLOWORLD;
