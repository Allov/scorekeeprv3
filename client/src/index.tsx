import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import StyledApp from './components/StyledApp';
import registerServiceWorker from './registerServiceWorker';
import configureStore from './store';

const store = configureStore({});

ReactDOM.render(
  <Provider store={store}>
    <StyledApp />
  </Provider>,
  document.getElementById('root') as HTMLElement
);

registerServiceWorker();
