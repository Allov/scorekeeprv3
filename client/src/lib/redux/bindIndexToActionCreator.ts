interface ITransformed {
  [key:string]: {};
}

const transformObjectValues = (obj: any, fn: (a: any) => any) => {
  const transformed: ITransformed = {};
  Object.keys(obj).forEach(key => {
    transformed[key] = fn(obj[key]);
  });
  return transformed;
}

const bindActionCreator = (actionCreator: (...args: any[]) => any, index: number) =>
  (...args: any[]) => Object.assign(actionCreator(...args), { index });

const bindActionCreatorMap = (creators: any[], index: number) =>
  transformObjectValues(creators, (actionCreator: (...args: any[]) => any) => bindActionCreator(actionCreator, index));

const bindIndexToActionCreators = (actionCreators: any, index: number) => {
  return typeof actionCreators === 'function'
    ? bindActionCreator(actionCreators, index)
    : bindActionCreatorMap(actionCreators, index);
}

export default bindIndexToActionCreators;
