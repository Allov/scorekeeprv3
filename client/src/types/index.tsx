import { RouterState } from 'connected-react-router';

// todo: split this... probably.

// state types
export interface IStore {
  configuration?: IConfiguration;
  modal?: any;
  pages?: IPage[];
  router?: RouterState;
}

export interface IConfiguration {
  theme: string;
}

// presentation types
export interface IPage {
  title: string;
  data: any;
}

export interface ILineup {
  title: string;
  items: ICard;
}

export interface ICard {
  title: string;
  description?: string; // todo: html
  link: string;
}
