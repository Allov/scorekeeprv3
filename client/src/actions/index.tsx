import * as constants from '../constants';

export interface ICreateGame {
  type: constants.CREATE_GAME;
  name: string;
}

export interface IFetchHelloworld {
  type: constants.FETCH_HELLOWORLD;
}

export type ScorekeeprActions = ICreateGame | IFetchHelloworld;

export function createGame(name: string) : ICreateGame {
  return {
    name,
    type: constants.CREATE_GAME,
  }
}

export function fetchHelloworld() : IFetchHelloworld {
  return {
    type: constants.FETCH_HELLOWORLD,
  }
}
