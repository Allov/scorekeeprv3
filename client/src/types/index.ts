import { RouterState } from 'connected-react-router';
import { IGameAdminPage } from '../containers/Pages/GameAdmin/reducer';
import { Notifications } from './constants';

// todo: split this... probably.

// state types
export interface IStore {
  configuration?: IConfiguration;
  notifications?: INotification[];
  modal?: any;
  pages?: IPages;
  router?: RouterState;
}

export interface IConfiguration {
  theme: string;
}

export interface INotification {
  type: Notifications;
  message: string;
}

// this needs to be shared between client and backend.
export interface IGameInput {
  name: string;
  userId?: string;
}

export interface IGame {
  id: string;
  name: string;
  shareId: string;
  createdBy: string;
}

export interface IUser {
  username: string;
}

// mm
export interface IPages {
  gameAdmin: IGameAdminPage;
}

// presentation types
export interface IPage {
  title: string;
  data: any;
}

export interface ILineup {
  title: string;
  items: ICard;
}

export interface ICard {
  title: string;
  description?: string; // todo: html
  link: string;
}
