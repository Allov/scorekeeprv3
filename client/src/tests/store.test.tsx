import { compose } from 'redux';
import { IStore } from 'src/types';
import configureStore from '../store';

it('should configure the redux store correctly', () => {
  const state: IStore = {
    configuration: {
      theme: 'DEFAULT',
    },
    modal: {},
    pages: [],
  };

  const store = configureStore(state);
  expect(store.getState()).toEqual(state);
});

it('should configure the redux store correctly with developer tool', () => {
  const state: IStore = {
    configuration: {
      theme: 'DEFAULT',
    },
    modal: {},
    pages: [],
  };

  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ = compose;

  const store = configureStore(state);
  expect(store.getState()).toEqual(state);
});
