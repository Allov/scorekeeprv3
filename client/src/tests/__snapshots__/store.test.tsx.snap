// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure the redux store correctly 1`] = `
Object {
  "configuration": Object {
    "theme": "day",
  },
  "notifications": Array [],
  "pages": Object {
    "gameAdmin": Object {
      "data": Object {},
      "game": Object {
        "createdBy": "",
        "currentRound": 1,
        "id": "",
        "name": "",
        "players": Array [],
        "shareId": "",
      },
      "title": "Game Admin",
    },
  },
  "router": Object {
    "action": "POP",
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
  },
}
`;

exports[`should configure the redux store correctly with developer tool 1`] = `
Object {
  "configuration": Object {
    "theme": "day",
  },
  "notifications": Array [],
  "pages": Object {
    "gameAdmin": Object {
      "data": Object {},
      "game": Object {
        "createdBy": "",
        "currentRound": 1,
        "id": "",
        "name": "",
        "players": Array [],
        "shareId": "",
      },
      "title": "Game Admin",
    },
  },
  "router": Object {
    "action": "POP",
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
  },
}
`;
